File: leetcode/js/check-if-word-equals-summation-of-two-words.js
/**
 * @param {string} firstWord
 * @param {string} secondWord
 * @param {string} targetWord
 * @return {boolean}
 */
var isSumEqual = function (firstWord, secondWord, targetWord) {
  const convertWordToNum = (word) => {
    return parseInt(
      [...word].map((c) => String(c.charCodeAt(0) - 97)).join(""),
      10
    );
  };

  const firstNum = convertWordToNum(firstWord);
  const secondNum = convertWordToNum(secondWord);

  const targetNum = convertWordToNum(targetWord);

  return firstNum + secondNum === targetNum;
};

File: leetcode/rust/check-if-word-equals-summation-of-two-words.rs
impl Solution {
    pub fn is_sum_equal(first_word: String, second_word: String, target_word: String) -> bool {
        let first_num = Self::convert_word_to_num(first_word);
        let second_num = Self::convert_word_to_num(second_word);

        let target_num = Self::convert_word_to_num(target_word);

        first_num + second_num == target_num
    }

    fn convert_word_to_num(word: String) -> i32 {
        word.bytes()
            .map(|b| (b - 97).to_string())
            .collect::<String>()
            .parse()
            .unwrap()
    }
}
